### PR TITLE
Renamed filter to keep/discard - Issue 240

### PR DESCRIPTION
**Pull request by javidoc00**
Renamed stream_filter to keep/discard.

### CODE REVIEW

File: doc/stream-filter.md [hunk 1]
```diff
@@ -30,7 +30,11 @@ predicate `pred` is any operation, that given a value `x` of type `T`, makes the
 following valid:
 
 ~~~{.cpp}
```
Comment (jdgarciauc3m) [inline doc/stream-filter.md:32]: Remove lines with comments and leave the example as

~~~c++
if (predicate(item)) { /* ... */ }
if (!predicate(item)) { /* ... */ }
~~~

File: doc/stream-filter.md [hunk 2]
```diff
@@ -93,26 +97,26 @@ also responsible for consuming the output values.
...
-  grppi::stream_filter(exec, [](auto x) { return x.lenght()>4; }),
+  grppi::keep(exec, [](auto x) { return x.lenght()>4; }),
   stageC
   );
 ~~~
 ---
 **Note**: For brevity we do not show here the details of other stages.
 
-For composing complex patterns, the **filter()** function may be used to create
+For composing complex patterns, the **keep()** and **discard()** functions may be used to create
```
Comment (jdgarciauc3m) [inline doc/stream-filter.md]: Instead of **keep()** use `keep()`
Instead of **discard()** use `discard()`

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]
Make sure that there is no reference to farm due to copy/paste.

In stand-alone filter section when enumerating elements statet that the element **Predicate** is 

> A *Predicate* for keeping or discarding elements.

In general, instead of talking about "filtering out" use "discarding"

**Comment by javidoc00** [comment]
Fixed comments

**Review by jdgarciauc3m — APPROVED** [review]
OK

**Comment by jdgarciauc3m** [comment]
Solves issue #240

**Pull request merged**